### PR TITLE
New feature #12120: direct action by php-cli

### PR DESCRIPTION
**Pull request by Shnoulle**
- Use index : default action
- param target mandatory
- 2 optionnal parameters function and option

### CODE REVIEW

**Comment by Shnoulle** [comment]
Usage  : cron event don't have target. If we add target now : old cron event don't use it => we break all old plugin using cron event.

Then need another plugin event with mandatory target type. Plugin can use target or not (like newDirectRequest : any plugin can die even if not target).

Add 2 extra parameter function + option : Yii don't accept any number of parameters with -- , user must use $_SERVER['argv'] for all other parameters (or CConsoleCommand->resolveRequest )

**Comment by Shnoulle** [comment]
I merge it monday if there are no issues

File: application/commands/PluginCommand.php [hunk 1]
```diff
@@ -15,13 +15,36 @@ class PluginCommand extends CConsoleCommand
...
         {
             $pm = \Yii::app()->pluginManager;
             $event = new PluginEvent('cron');
             $event->set('interval', $interval);
             $pm->dispatchEvent($event);
         }
+
+        /**
+         * Call direct by command (it's default)
+         * @plugin string : the plugin name, target of action
```
Comment (olleharstedt) [inline application/commands/PluginCommand.php]: What's up with the PHPdoc block? Should be
```
@param string $target
```
etc.

Comment (Shnoulle) [inline application/commands/PluginCommand.php]: yes, should be :)

**Pull request merged**